lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,432 @@
+
+        try:
+            # fetch the user data
+            user = (
+                Database.db_session.query(UserModel)
+                .filter_by(username=username)
+                .first()
+            )
+
+        except Exception:
+            self.logger.exception("Exception occurred during user login")
+            abort(500, message="INTERNAL ERROR")
+
+        if not user:
+            self.logger.warning(
+                "No user found in the db for Username: {} while login".format(username)
+            )
+            abort(403, message="No such user, please register first")
+
+        # Validate the password
+        if not check_password_hash(user.password, password):
+            self.logger.warning(
+                "Wrong password for user: {} while login".format(username)
+            )
+            abort(401, message="Wrong password")
+
+        try:
+            auth_token = encode_auth_token(self.token_expire_duration, user.id)
+        except (
+            jwt.InvalidIssuer,
+            jwt.InvalidIssuedAtError,
+            jwt.InvalidAlgorithmError,
+            jwt.PyJWTError,
+        ):
+            self.logger.error(
+                "Could not encode the JWT auth token for user: {} while login".format(
+                    username
+                )
+            )
+            # return make_response(jsonify(response_object))
+            abort(
+                500, message="INTERNAL ERROR",
+            )
+
+        """
+        Add the new auth token to the database for later access
+        """
+        try:
+            token = ActiveTokens(token=auth_token.decode())
+            user.auth_tokens.append(token)
+
+            # insert the user
+            Database.db_session.add(token)
+            Database.db_session.commit()
+            self.logger.info("New auth token registered for user {}".format(user.email))
+
+        except Exception:
+            self.logger.exception("Exception while registering a user")
+            abort(500, message="INTERNAL ERROR")
+
+        response_object = {
+            "status": "success",
+            "message": "Successfully logged in.",
+            "auth_token": auth_token.decode(),
+        }
+        return make_response(jsonify(response_object), 200)
+
+    @auth.login_required
+    def delete(self):
+        """
+        Post request for logging out a user.
+        This requires a JWT auth token in the header field
+
+        Required headers include
+            Authorization:   JWT token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 200/status_code_int,
+                    "message": "Successfully logged out."/"failure message",
+                }
+        """
+        _, auth_token = request.headers.get("Authorization").split()
+        user = auth.current_user()
```

Comment:
The reason I didnt do that is because `@auth.login_required` decorator is defined on top of this method, therefore if we reach this line of code we are certain that the auth token is present in the header in a proper format, and so no need to re-do the auth check again.